packages/labs/src/components/popover/popover2.tsx

```diff
@@ -288,9 +289,10 @@ export class Popover2 extends AbstractComponent<IPopover2Props, IPopover2State>
         );
 
         const children = this.understandChildren();
-        const targetTabIndex = this.props.openOnTargetFocus && this.isHoverInteractionKind() ? 0 : undefined;
+        const targetTabIndex = this.props.openOnTargetFocus && isHoverInteractionKind ? 0 : undefined;
         const target: JSX.Element = React.cloneElement(children.target, {
             // force disable single Tooltip child when popover is open (BLUEPRINT-552)
+            className: classNames({ [Classes.ACTIVE]: isOpen && !isHoverInteractionKind }),
```

Comment:
Need to make sure we don't overwrite custom target classes that might have been provided. Let's move this to the `classNames` composition on [L282](https://github.com/palantir/blueprint/pull/1872/files#diff-affb716df70db56e0ecd5eecaae135abR283):

```tsx
targetProps.className = classNames(
    Classes.POPOVER_TARGET,
    {
        [Classes.ACTIVE]: isOpen && !isHoverInteractionKind,
        [Classes.POPOVER_OPEN]: isOpen,
    },
    className,
);
```

Reply:
@cmslewis `targetProps.className` or `target.className`?

Reply:
`targetProps`. Here's the code that's already present:

![image](https://user-images.githubusercontent.com/443450/33693880-19397c18-daaa-11e7-9f19-848f1ad48470.png)